http/client/src/main/java/org/wildfly/security/http/client/mechanism/bearer/ElytronHttpClientBearerAuthMechanism.java

```diff
@@ -0,0 +1,26 @@
+package org.wildfly.security.http.client.mechanism.bearer;
+
+import org.wildfly.security.http.client.utils.ElytronHttpClientCredentialUtils;
+
+import java.net.URI;
+import java.net.http.HttpRequest;
+
+public class ElytronHttpClientBearerAuthMechanism {
+
+    private static final String AUTHORIZATION = "Authorization";
+    private static ElytronHttpClientCredentialUtils elytronHttpClientCredentialProvider = new ElytronHttpClientCredentialUtils();
+
+    public static HttpRequest evaluateMechanism(URI uri) {
+        String token = elytronHttpClientCredentialProvider.getToken(uri);
+        HttpRequest request = HttpRequest
+                .newBuilder()
+                .uri(uri)
+                .header(AUTHORIZATION, getBearerHeader(token))
+                .build();
+        return request;
+    }
+
+    private static String getBearerHeader(String token){
```

Comment:
@keshav-725 Just a minor, here and in the other classes the curly bracket sometimes have missing space before it. I think by default if you use intellij idea reformatting (From the main menu, select Code | Reformat Code or press Ctrl+Alt+L by default) in all your files it should format it properly